packages/flutter_tools/test/widget_preview_scaffold.shard/widget_preview_scaffold/lib/src/dtd/dtd_services.dart

```diff
@@ -92,12 +92,14 @@ class WidgetPreviewScaffoldDtdServices with DtdEditorService {
   /// preferences map.
   ///
   /// Returns null if [key] is not in the map.
-  Future<String?> getPreference(String key) async {
+  Future<Object?> getPreference(String key) async {
     try {
-      final response = StringResponse.fromDTDResponse(
-        (await _call(kGetPreference, params: {'key': key}))!,
-      );
-      return response.value;
+      final response = await _call(kGetPreference, params: {'key': key});
+      return switch (response!.type) {
```

Comment:
What about using `?` instead of `?` and have this fall into the `StateError` case so that if it happens the error message is a bit clearer?

Reply:
SGTM!